crates/bevy_debug_overlays/src/lib.rs

```diff
@@ -0,0 +1,198 @@
+            },
+            ..default()
+        })
+        .with_children(|parent| {
+            parent
+                .spawn(NodeBundle {
+                    background_color: Color::rgba(0.10, 0.10, 0.10, 0.8).into(),
+                    style: Style {
+                        padding: UiRect::all(Val::Px(8.0)),
+                        flex_direction: FlexDirection::Column,
+                        ..default()
+                    },
+                    ..default()
+                })
+                .with_children(|parent| {
+                    parent.spawn(TextBundle::from_sections([
+                        TextSection::new(
+                            "GPU Time",
+                            TextStyle {
+                                font_size: 16.0,
+                                color: Color::WHITE,
+                                ..default()
+                            },
+                        ),
+                        TextSection::new(
+                            " (ms)  Mean     Min      Max\n",
+                            TextStyle {
+                                font_size: 10.0,
+                                color: Color::WHITE,
+                                ..default()
+                            },
+                        ),
+                    ]));
+                    parent.spawn(NodeBundle::default()).with_children(|parent| {
+                        let style = TextStyle {
+                            font_size: 12.0,
+                            color: Color::WHITE,
+                            ..default()
+                        };
+                        parent.spawn((
+                            GpuTimerLabelMarker,
+                            TextBundle::from_section("", style.clone()),
+                        ));
+                        parent.spawn((GpuTimerDurationMarker, TextBundle::from_section("", style)));
+                    });
+                });
+        });
+}
+
+fn draw_gpu_time_overlay(
+    gpu_timers: Res<GpuTimerScopes>,
+    mut labels: Query<&mut Text, (With<GpuTimerLabelMarker>, Without<GpuTimerDurationMarker>)>,
+    mut durations: Query<&mut Text, (With<GpuTimerDurationMarker>, Without<GpuTimerLabelMarker>)>,
+) {
+    let mut labels = labels.single_mut();
+    let mut durations = durations.single_mut();
+    let labels = &mut labels.sections[0].value;
+    let durations = &mut durations.sections[0].value;
+    labels.clear();
+    durations.clear();
+
+    let gpu_timers = gpu_timers.get();
+    let mut aggregated_timers: Vec<AggregatedGpuTimer> = Vec::new();
+
+    for frame in gpu_timers.iter() {
+        for timer in frame {
+            let timer_duration = (timer.time.end - timer.time.start) * 1000.0;
+            match aggregated_timers
+                .iter_mut()
+                .find(|a| a.label == timer.label)
+            {
+                Some(a) => {
+                    a.mean_duration += timer_duration / 20.0;
```

Comment:
I think this should at least be a constant that `GpuTimerScopes` also uses. User configurable would be even better.